plugins/woocommerce/src/Internal/ProductAttributesLookup/DataRegenerator.php

```diff
@@ -165,6 +167,41 @@ private function initialize_table_and_data() {
 		return true;
 	}
 
+	/**
+	 * Create the indices for the lookup table unless they exist already.
+	 */
+	public function maybe_create_table_indices() {
+		global $wpdb;
+
+		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
+		$wpdb->query(
+			"
+IF (SELECT COUNT(1)
+	FROM INFORMATION_SCHEMA.STATISTICS
+	WHERE table_schema=' " . $wpdb->dbname . "'
+	AND table_name='" . $this->lookup_table_name . "'
+	AND index_name='product_or_parent_id_term_id')=0
+THEN
+	ALTER TABLE wp_wc_product_attributes_lookup
+	ADD INDEX product_or_parent_id_term_id (product_or_parent_id, term_id);
+END IF;"
+		);
+
+		$wpdb->query(
+			"
+IF (SELECT COUNT(1)
```

Comment:
Is an `IF THEN` statement (as opposed to the `IF()` *function)* definitely allowed outside a stored procedure? I seem to keep hitting errors with this:

```
You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'IF
```

Not sure if this is a version-specific thing (for ref, I was testing this with MySQL 8.0.27) or a configuration issue, or something else. What do you think?

Reply:
I'm using MariaDB and it works, but [the MySQL documentation](https://dev.mysql.com/doc/refman/5.6/en/if.html) seems to suggest that indeed it's intended for stored procedures only. So I have changed it by creating a temporary stored procedure, executing it and then deleting it.